lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -693,8 +693,8 @@ void SpecialFunctionHandler::handleMakeSymbolic(ExecutionState &state,
     
     const ObjectState *old = it->first.second;
     ExecutionState *s = it->second;
-    
-    if (old->readOnly) {
+
+    if (old->readOnly || isConstant(mo->allocSite)) {
```

Comment:
It also makes me wonder why `old->readOnly` is not set to `true` given that the `ObjectState` was created for a `MemoryObject` that is for an allocation site marked as `const`. Surely there's a bug elsewhere?